src/hue_api.ts

```diff
@@ -47,3 +48,29 @@ export const request = (method, path, body) => new Promise((resolve, reject) =>
   req.write(JSON.stringify(body));
   req.end();
 });
+
+export const getRoomsFromGroups = (groups: Groups) => {
```

Comment:
This feels like it's maybe a level of functionality one level higher up than hue_api.ts. HueAPI is a pretty thin wrapper around the bridge, barely translating anything; getRoomsFromGroups feels a little more complex. Maybe should be elsewhere perhaps a new file. "dashboard.ts" maybe?

Reply:
A great idea! ❤️